security/security_profiles_operator/spo-enabling.adoc

```diff
@@ -0,0 +1,13 @@
+:_content-type: ASSEMBLY
+[id="spo-enabling"]
+= Enabling the Security Profiles Operator
+include::_attributes/common-attributes.adoc[]
+:context: spo-enabling
+
+toc::[]
+
+Before you can use the Security Profiles Operator, you must ensure it is deployed in the cluster.
+
+include::modules/spo-installing.adoc[leveloffset=+1]
```

Comment:
The current installation only contains the installation from GUI.
Could we add `install Security Profiles Operator using the CLI`? Thanks.
I think most customers will be interested in this part.
@jhrozek  @sheriff-rh

Reply:
Hmm, on the one hand I agree that we should probably include the CLI installation instructions, on the other hand I don't know if we still have the time and I'd defer to @sheriff-rh on that.

Below is what I think might be good enough content adapted from CO docs:
```
 
Prerequisites
    You must have admin privileges.
 
Procedure
1. Define a Namespace object:
 
Example namespace-object.yaml
 
apiVersion: v1
kind: Namespace
metadata:
  labels:
    openshift.io/cluster-monitoring: "true"
  name: openshift-security-profiles
 
2. Create the Namespace object:
 
$ oc create -f namespace-object.yaml
 
3. Define an OperatorGroup object:
Example operator-group-object.yaml
 
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: security-profiles-operator
  namespace: openshift-security-profiles
 
4. Create the OperatorGroup object:
 
$ oc create -f operator-group-object.yaml


5. Define a Subscription object:
Example subscription-object.yaml
 
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: compliance-operator-sub
  namespace: openshift-compliance
spec:
  channel: release-alpha-rhel-8
  installPlanApproval: Automatic
  name: security-profiles-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
 
6. Create the Subscription object:
 
$ oc create -f subscription-object.yaml
```
@xiaojiey would you have time to test the above section out? I think the only modification you would have to do now when SPO is not yet shipped as part of  `redhat-operators` would be to change `source` in the `Subscription` from `redhat-operators` to your `CatalogSource` name...

Reply:
@jhrozek Geneally it is Okay. But there are minor issues:
1. for the namespace, better to add label `pod-security.kubernetes.io/enforce: privileged` as below:
```
apiVersion: v1
kind: Namespace
metadata:
  labels:
    openshift.io/cluster-monitoring: "true"
    pod-security.kubernetes.io/enforce: privileged
  name: openshift-security-profiles
```
2. For the sub, need to update the name and namespace(pointing to openshift-compliance could be confusing):
```
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: security-profiles-operator-sub
  namespace: openshift-security-profiles
spec:
  channel: release-alpha-rhel-8
  installPlanApproval: Automatic
  name: security-profiles-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
```

Reply:
Confirmed that the privileged label is not needed and OLM adds it on its own. So this install manifest:
```
apiVersion: v1
kind: Namespace
metadata:
    name: openshift-security-profiles
labels:
  openshift.io/cluster-monitoring: "true"
```
becomes this namespace object after the CSV is installed:
```
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    openshift.io/sa.scc.mcs: s0:c26,c15
    openshift.io/sa.scc.supplemental-groups: 1000680000/10000
    openshift.io/sa.scc.uid-range: 1000680000/10000
  creationTimestamp: "2023-01-12T12:49:23Z"
  labels:
    kubernetes.io/metadata.name: openshift-security-profiles
    pod-security.kubernetes.io/enforce: privileged
    pod-security.kubernetes.io/enforce-version: v1.24
    security.openshift.io/scc.podSecurityLabelSync: "true"
  name: openshift-security-profiles
  resourceVersion: "94965"
  uid: a668db67-e349-4d35-a6b8-30d76d373215
```

Reply:
> Nice catch, now you know I took the compliance operator docs as a base :-)

I'm also stealing from CO docs for creating the file haha

Reply:
Unrelated: I have created a [task](https://issues.redhat.com/browse/OSDOCS-4790) to get the FIO/CO/SPO installing via CLI pages updated. These procedures should be more similar than they are.  I will tackle this at a later date.